designer/listener-api/src/main/scala/pl/touk/nussknacker/ui/listener/services/RepositoryScenarioWithDetails.scala

```diff
@@ -0,0 +1,53 @@
+package pl.touk.nussknacker.ui.listener.services
+
+import pl.touk.nussknacker.engine.api.deployment.ProcessAction
+import pl.touk.nussknacker.engine.api.process.{ProcessId, ProcessIdWithName, ProcessName, VersionId}
+import pl.touk.nussknacker.engine.api.{ProcessVersion => EngineProcessVersion}
+import pl.touk.nussknacker.restmodel.process.ProcessingType
+import pl.touk.nussknacker.restmodel.scenariodetails
+
+import java.time.Instant
+
+// TODO we should split ScenarioDetails and ScenarioShape (json)
+final case class RepositoryScenarioWithDetails[ScenarioShape](
```

Comment:
Yeah, it should be in the repository package because it is an internal thing, but because we have this PullProcessRepository, we need to move it to listener-api. It never was a DTO, we always passed this object from repository directly to FE with some transformations in the middle. I only explicitly exposed this thing. I think that proposed by you ScenarioWithDetailsDB` won't solve the problem, it is the same but we fix that our repositories are db repositories. Where is the benefit? `Raw`? When do things stop being "raw"? Regarding the layers concept, I don't understand what exactly you propose. What do you want to move where?